src/reducers/mainReducer.js

```diff
@@ -14,6 +14,8 @@ const initialState = {
     results: [],
     isLoading: false,
     resultFilter: '',
+    showModal: false,
```

Comment:
Since this is a boolean, change the name to something like `isModalShown`. It's clear here, but since you followed through with the name in the components and containers, it is confusing there since it seems like a function.